adapters/eea-direct.js

```diff
@@ -0,0 +1,177 @@
+  // filter for only country of interest's records
+  filter(metadata, (record, truth) => {
+    truth(record[0] === country);
+  }, (countryMetadata) => {
+    // map filtered records to be a list of objs w stationId/coordinates
+    map(countryMetadata, (record, done) => {
+      const station = {
+        stationId: record[5],
+        coordinates: {
+          latitude: record[15],
+          longitude: record[14]
+        }
+      };
+      done(null, station);
+    }, (err, mappedRecords) => {
+      if (err) {
+        return callback(null, []);
+      }
+      callback(null, uniqBy(mappedRecords, 'stationId'));
+    });
+  });
+};
+
+// makes requests to get country's pollutant data.
+const makeTaskRequests = (source) => {
+  const pollutantRequests = acceptableParameters.map((pollutant) => {
+    pollutant = pollutant.toUpperCase();
+    return (done) => {
+      const url = source.url.replace('<pollutant>', pollutant);
+      request.get({
+        url: url
+      }, (err, res, body) => {
+        if (err || res.statusCode !== 200) {
+          return done(null, []);
+        }
+        done(null, parse(body).data.slice(1, -1));
+      });
+    };
+  });
+  return (done) => {
+    parallel(
+      pollutantRequests,
+      (err, response) => {
+        if (err) {
+          done(null, []);
+        }
+        done(null, [].concat.apply([], response));
+      }
+    );
+  };
+};
+
+// formats data to match openAQ standard
+const formatData = (data, source, cb) => {
+  const coordinates = data[0];
+  const records = data[1];
+  map(records, (record, cb) => {
+    let measurement = {
+      parameter: record[5],
+      date: makeDate(record[16]),
+      coordinates: makeCoordinates(coordinates, record[11]),
+      value: record[19],
+      unit: record[record.length - 1],
+      attribution: [{
+        name: 'EEA',
+        url: source.sourceUrl
+      }],
+      averagingPeriod: {
+        unit: 'hours',
+        value: makeAvgPeriod(record.slice(15, 17)).toString()
+      }
+    };
+    // apply unit conversion to generated record
+    cb(null, convertUnits([measurement])[0]);
+  }, (err, mappedRecords) => {
+    if (err) {
+      return cb(null, []);
+    }
+    cb(null, mappedRecords);
+  });
+};
+
+const makeCoordinates = (coordinatesList, stationId) => {
+  return coordinatesList.filter((coordinates) => {
+    return coordinates.stationId === stationId;
+  }).map((station) => {
+    return {
+      latitude: parseFloat(station.coordinates.latitude),
+      longitude: parseFloat(station.coordinates.longitude)
+    };
+  })[0];
+};
+
+const makeAvgPeriod = (delta) => {
+  const latestTime = moment.tz(delta[1], 'YYYY-MM-DD hh:mm:ss', 'Europe/Berlin');
+  const earliestTime = moment.tz(delta[0], 'YYYY-MM-DD hh:mm:ss', 'Europe/Berlin');
+  return moment(latestTime).diff(earliestTime, 'hours');
+};
+
+const makeDate = (date) => {
+  let timeZone = date.slice(date.length - 6, date.length);
+  switch (timeZone) {
```

Comment:
Did some investigation around how EEA handles time. See the comment here: https://github.com/openaq/openaq-fetch/issues/298#issuecomment-318382840